src/metabase/driver/postgres.clj

```diff
@@ -1006,23 +1006,58 @@
     ::upload/datetime                 [:timestamp]
     ::upload/offset-datetime          [:timestamp-with-time-zone]))
 
+(defmethod driver/upload-promotion-allowlist :postgres
+  [_driver]
+  {::upload/int     #{::upload/float}
+   ::upload/boolean #{::upload/int
+                      ::upload/float}})
+
 (defmethod driver/create-auto-pk-with-append-csv? :postgres
   [driver]
   (= driver :postgres))
 
-(defmethod sql-jdbc.sync/alter-columns-sql :postgres
-  [driver table-name column-definitions]
+(defn- alter-column-using-hsql-expr
+  "In postgres some ALTER COLUMN statements generated by replacing or appending csv files
+  will result in an error, e.g. boolean columns cannot be changed to bigint.
+
+  This function returns a honey expr suitable for use with the USING keyword to tell postgres how to transform
+  values of the old type to the new, to avoid an error.
+
+  It returns nil if no such expression has been defined for the pair of types. In this case, the caller should
+  generate the ALTER COLUMN statement without a USING."
+  [column old-type new-type]
+  (condp = [old-type new-type]
```

Comment:
nit: I think this can just be `case` statement

Reply:
e39fe33326d7